cmd/installer/subcommands/installer/command.go

```diff
@@ -0,0 +1,181 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package installer contains the installer subcommands
+package installer
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/DataDog/datadog-agent/cmd/installer/command"
+	"github.com/DataDog/datadog-agent/pkg/fleet/installer"
+	"github.com/DataDog/datadog-agent/pkg/fleet/telemetry"
+	"github.com/spf13/cobra"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Commands returns the installer subcommands.
+func Commands(_ *command.GlobalParams) []*cobra.Command {
+	return []*cobra.Command{installCommand(), removeCommand(), installExperimentCommand(), removeExperimentCommand(), promoteExperimentCommand(), garbageCollectCommand()}
+}
+
+type installerCmd struct {
+	installer.Installer
+	t    *telemetry.Telemetry
+	ctx  context.Context
+	span ddtrace.Span
+}
+
+func (i *installerCmd) Stop(err error) {
+	i.span.Finish(tracer.WithError(err))
+	if i.t != nil {
+		err := i.t.Stop(context.Background())
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to stop telemetry: %v\n", err)
+		}
+	}
+}
+
+func newInstallerCmd(operation string) *installerCmd {
+	span, ctx := newSpan(operation)
+	registry := os.Getenv("DD_INSTALLER_REGISTRY")
+	registryAuth := os.Getenv("DD_INSTALLER_REGISTRY_AUTH")
+	var installerOpts []installer.Options
+	if registry != "" {
+		installerOpts = append(installerOpts, installer.WithRegistry(registry))
+	}
+	if registryAuth != "" {
+		installerOpts = append(installerOpts, installer.WithRegistryAuth(installer.RegistryAuth(registryAuth)))
+	}
+	i := installer.NewInstaller(installerOpts...)
+	return &installerCmd{
+		Installer: i,
+		t:         newTelemetry(),
+		ctx:       ctx,
+		span:      span,
+	}
+}
+
+func newTelemetry() *telemetry.Telemetry {
+	apiKey := os.Getenv("DD_API_KEY")
+	site := os.Getenv("DD_SITE")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ Question:** ‏Shouldn't we have a default?